src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1255,6 +1257,36 @@ class spell_pri_guardian_spirit : public AuraScript
     }
 };
 
+// 421558 - Heaven's Wrath
+class spell_pri_heavens_wrath : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_HEAVENS_WRATH,
+            SPELL_PRIEST_ULTIMATE_PENITENCE
+        });
+    }
+
+    void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* caster = eventInfo.GetActor();
+        if (!caster)
+            return;
+
+        SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(SPELL_PRIEST_ULTIMATE_PENITENCE, GetCastDifficulty());
```

Comment:
missing return here

Reply:
wrong advice, ignore

@Nyr97 - its validated

Reply:
dont we need ti assert in that case?

Reply:
If the validation fails script is not loaded

Reply:
oh I understand